docs/API/API-socket-options.md

```diff
@@ -1296,16 +1300,22 @@ This value is only significant when [`SRTO_TSBPDMODE`](#SRTO_TSBPDMODE) is enabl
 **Default value**: 120 ms in Live mode, 0 in File mode (see [`SRTO_TRANSTYPE`](#SRTO_TRANSTYPE)).
 
 The latency value defines the **minimum** receiver buffering delay before delivering an SRT data packet
-from a receiving SRT socket to a receiving application. The provided value is used in the connection establishment (handshake exchange) stage
-to fix the end-to-end latency of the transmission. The effective end-to-end latency `L` will be fixed
-as the network transmission time of the final handshake packet (~1/2 RTT) plus the **negotiated** latency value `Ln`.
-Data packets will stay in the receiver buffer for at least `L` microseconds since the timestamp of the
-packet, independent of the actual network transmission times (RTT variations) of these packets.
+from a receiving SRT socket to a receiving application.
 
 The actual value of the receiver buffering delay `Ln` (the negotiated latency) used on a connection
 is determined by the negotiation in the connection establishment (handshake exchange) phase as the maximum of the
 `SRTO_RCVLATENCY` value and the value of [`SRTO_PEERLATENCY`](#SRTO_PEERLATENCY) set by the peer.
 
+The general idea for the latency mechanism is to keep the time distance between two consecutive
+received packets the same as the time when these same packets were scheduled for sending by the
+sender application (or per the time explicitly declared when sending - see
+[`srt_sendmsg2`](API-functions.md#srt_sendmsg2) for details). This makes the packets, that have arrived
+earlier than their delivery time, kept in the receiver buffer until this time comes. This should
+compensate any jitter in the network and an extra delay needed for a packet retransmission.
+
+For the detailed information on how the latency setting influences the actual packet delivery time and
+how this time is defined, refer to the [latency documentation](../features/latency.md).
```

Comment:
```suggestion
[`srt_sendmsg2`](API-functions.md#srt_sendmsg2) for details). This keep any packets that have arrived
earlier than their delivery time in the receiver buffer until their scheduled time comes. This should
compensate for any jitter in the network and provides an extra delay needed for a packet retransmission.

For detailed information on how the latency setting influences the actual packet delivery time and
how this time is defined, refer to the [latency documentation](../features/latency.md).
```

Reply:
"This keep"? Not keeps?

And although theoretically "scheduled time" would be of a correct meaning, it can be ambiguous and kinda misleading, as "scheduled" is used mainly for sender in this description.  Maybe this can be simply shifted to apply to this delivery time additionally and this place could be kept with a stub: "This keeps any packets, that have arrived earlier than their scheduled delivery time, in the receiver buffer, until this time comes". Or maybe "inteded delivery time" could be a bit better here.

Reply:
```suggestion
[`srt_sendmsg2`](API-functions.md#srt_sendmsg2) for details). This keeps any packets that have arrived
earlier than their delivery time in the receiver buffer until their retrieval time comes. This should
compensate for any jitter in the network and provides an extra delay needed for a packet retransmission.
	
For detailed information on how the latency setting influences the actual packet delivery time and
how this time is defined, refer to the [latency documentation](../features/latency.md).
```

Reply:
Yup, keeps, not keep . . . good catch!
Your other change seems reasonable to me.  :-)